pkg/cmd/webhooks/forward_test.go

```diff
@@ -0,0 +1,191 @@
+
+	f := &cmdutil.Factory{
+		IOStreams: ios,
+		Config: func() (config.Config, error) {
+			return config.NewBlankConfig(), nil
+		},
+		HttpClient: func() (*http.Client, error) {
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+
+			return &http.Client{Transport: tr}, nil
+		},
+	}
+
+	wsServer := getWSServer()
+	defer wsServer.Close()
+
+	ghAPIServer := getGHAPIServer(wsServer.URL)
+	defer ghAPIServer.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	webhookRcvServer, forwarded := getWebhookRcvServer(wg.Done)
+	defer webhookRcvServer.Close()
+	webhookRcvServerURL = webhookRcvServer.URL
+
+	maxRetries = 1
+	cmd := newCmdForward(f, nil)
+	event := "issues"
+	splitURL := strings.Split(webhookRcvServerURL, ":")
+	port := splitURL[len(splitURL)-1]
+	repo := "monalisa/smile"
+	host := strings.TrimPrefix(ghAPIServer.URL, "https://")
+
+	run := fmt.Sprintf("--event %s --port %s --repo %s --host %s", event, port, repo, host)
+	argv, err := shlex.Split(run)
+	assert.NoError(t, err)
+	cmd.SetArgs(argv)
+	cmd.SetIn(&bytes.Buffer{})
+	cmd.SetOut(&bytes.Buffer{})
+	cmd.SetErr(&bytes.Buffer{})
+
+	_, err = cmd.ExecuteC()
+	assert.NoError(t, err)
+	wg.Wait()
+	assert.Equal(t, "lol\n", string(forwarded.event.Body))
+	assert.Equal(t, forwarded.event.Header.Get("Someheader"), "somevalue")
+}
+
+type forwarded struct {
+	event localEvent
+	done  func()
+}
+
+type localEvent struct {
+	Body   []byte `json:"body"`
+	Header http.Header
+}
+
+func (w *forwarded) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	var event localEvent
+	err := json.NewDecoder(req.Body).Decode(&event)
+	if err != nil {
+		fmt.Errorf("failed to decode request: %s\n", err)
+	}
+	event.Header = http.Header{}
+	for h := range req.Header {
+		event.Header.Add(h, req.Header.Get(h))
+	}
+	w.event = event
+	_, err = res.Write([]byte("OK"))
+	if err != nil {
+		fmt.Errorf("failed to write response: %s\n", err)
+	}
+
+	w.done()
+}
+
+func getWebhookRcvServer(done func()) (*httptest.Server, *forwarded) {
+	s := &forwarded{done: done}
+	return httptest.NewServer(s), s
+}
+
+func getGHAPIServer(wsServerURL string) *httptest.Server {
+	ghAPIHandler := http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		wsURL := strings.Replace(wsServerURL, "http", "ws", 1)
+		ret := createHookResponse{
+			WsURL: wsURL,
+		}
+		err := json.NewEncoder(res).Encode(ret)
+		if err != nil {
+			fmt.Errorf("failed to write response: %s\n", err)
+		}
+	})
+	return httptest.NewTLSServer(ghAPIHandler)
+}
+
+func getWSServer() *httptest.Server {
+	wsHandler := http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		var upgrader = websocket.Upgrader{}
+		c, err := upgrader.Upgrade(res, req, nil)
+		if err != nil {
+			fmt.Errorf("failed to upgrade: %s\n", err)
+		}
+		defer c.Close()
+
+		header := http.Header{}
+		header.Add("Someheader", "somevalue")
+		msg := wsRequest{
+			Header: header,
+			Body:   []byte(`{"body": "bG9sCg=="}`),
```

Comment:
Totally up to you but i it makes things simpler: you just make the body "anything", like `Body: []byte("hello"),`  and then the forwardedHandler can just read the body and set it without having to decode the json on line 124. 

All the CLI does is take the body and give it to the customer's local server, so we don't need to assert its json or otherwise for now.